testnet/stacks-node/src/tests/signer/commands/block_commit.rs

```diff
@@ -0,0 +1,89 @@
+use super::context::{SignerTestContext, SignerTestState};
+use crate::tests::signer::v0::MultipleMinerTest;
+use madhouse::{Command, CommandWrapper};
+use proptest::prelude::{Just, Strategy};
+use std::sync::{Arc, Mutex};
+
+pub struct SubmitBlockCommitSecondaryMinerCommand {
+    miners: Arc<Mutex<MultipleMinerTest>>,
+}
+
+impl SubmitBlockCommitSecondaryMinerCommand {
+    pub fn new(miners: Arc<Mutex<MultipleMinerTest>>) -> Self {
+        Self { miners }
+    }
+}
+
+impl Command<SignerTestState, SignerTestContext> for SubmitBlockCommitSecondaryMinerCommand {
+    fn check(&self, state: &SignerTestState) -> bool {
+        println!(
+            "Checking: Submitting block commit miner 2. Result: {:?}",
+            state.is_secondary_miner_skip_commit_op
+        );
+        state.is_secondary_miner_skip_commit_op
+    }
+
+    fn apply(&self, _state: &mut SignerTestState) {
+        println!("Applying: Submitting block commit miner 2");
+
+        let (conf_1, _) = self.miners.lock().unwrap().get_node_configs();
+        let burnchain = conf_1.get_burnchain();
+        let sortdb = burnchain.open_sortition_db(true).unwrap();
+
+        self.miners.lock().unwrap().submit_commit_miner_2(&sortdb);
+    }
+
+    fn label(&self) -> String {
+        "SUBMIT_BLOCK_COMMIT_SECONDARY_MINER".to_string()
+    }
+
+    fn build(
+        ctx: Arc<SignerTestContext>,
+    ) -> impl Strategy<Value = CommandWrapper<SignerTestState, SignerTestContext>> {
```

Comment:
Spontaneously I would question the explicit tight-coupling to `proptest` here as it necessarily forces implementers of `Command`s to have direct knowledge of `proptest`.

I understand why you've done it this way, and I would suggest borrowing the `Rng` from the `proptest::TestRunner` and passing that in instead and letting the implementer do what they want with it, but I'm not sure how that works together with shrinking etc...